e2e-tests/src/ban.rs

```diff
@@ -124,3 +138,58 @@ pub fn check_ban_event<C: AnyConnection>(
 
     Ok(event)
 }
+
+pub fn check_session_count<C: AnyConnection>(
+    connection: &C,
+    seats: &CommitteeSeats,
+    reserved_validators: &[AccountId],
+    non_reserved_validators: &[AccountId],
+    start_session: &SessionIndex,
+    sessions_to_check: &SessionCount,
+    ban_session_threshold: &SessionCount,
+) -> anyhow::Result<()> {
+    let validators: Vec<_> = reserved_validators
+        .iter()
+        .chain(non_reserved_validators.iter())
+        .collect();
+
+    let mut expected_validator_session_count = HashMap::new();
+
+    for session in *start_session..*start_session + *sessions_to_check {
+        wait_for_at_least_session(connection, session)?;
```

Comment:
rather than waiting here, run this function after `start_session + sessions_to_check` have passed

Reply:
I have changed the test logic to check the storage values once all the checked sessions have passed.